openmc/data/neutron.py

```diff
@@ -0,0 +1,926 @@
+    Evaluation, SUM_RULES, get_head_record, get_tab1_record, get_evaluations)
+from .fission_energy import FissionEnergyRelease
+from .function import Tabulated1D, Sum, ResonancesWithBackground
+from .grid import linearize, thin
+from .njoy import make_ace
+from .product import Product
+from .reaction import Reaction, _get_photon_products_ace
+from . import resonance as res
+from . import resonance_covariance as res_cov
+from .urr import ProbabilityTables
+import openmc.checkvalue as cv
+from openmc.mixin import EqualityMixin
+
+
+# Fractions of resonance widths used for reconstructing resonances
+_RESONANCE_ENERGY_GRID = np.logspace(-3, 3, 61)
+
+
+class IncidentNeutron(EqualityMixin):
+    """Continuous-energy neutron interaction data.
+
+    This class stores data derived from an ENDF-6 format neutron interaction
+    sublibrary. Instances of this class are not normally instantiated by the
+    user but rather created using the factory methods
+    :meth:`IncidentNeutron.from_hdf5`, :meth:`IncidentNeutron.from_ace`, and
+    :meth:`IncidentNeutron.from_endf`.
+
+    Parameters
+    ----------
+    name : str
+        Name of the nuclide using the GND naming convention
+    atomic_number : int
+        Number of protons in the target nucleus
+    mass_number : int
+        Number of nucleons in the target nucleus
+    metastable : int
+        Metastable state of the target nucleus. A value of zero indicates ground
+        state.
+    atomic_weight_ratio : float
+        Atomic mass ratio of the target nuclide.
+    kTs : Iterable of float
+        List of temperatures of the target nuclide in the data set.
+        The temperatures have units of eV.
+
+    Attributes
+    ----------
+    atomic_number : int
+        Number of protons in the target nucleus
+    atomic_symbol : str
+        Atomic symbol of the nuclide, e.g., 'Zr'
+    atomic_weight_ratio : float
+        Atomic weight ratio of the target nuclide.
+    fission_energy : None or openmc.data.FissionEnergyRelease
+        The energy released by fission, tabulated by component (e.g. prompt
+        neutrons or beta particles) and dependent on incident neutron energy
+    mass_number : int
+        Number of nucleons in the target nucleus
+    metastable : int
+        Metastable state of the target nucleus. A value of zero indicates ground
+        state.
+    name : str
+        Name of the nuclide using the GND naming convention
+    reactions : collections.OrderedDict
+        Contains the cross sections, secondary angle and energy distributions,
+        and other associated data for each reaction. The keys are the MT values
+        and the values are Reaction objects.
+    resonances : openmc.data.Resonances or None
+        Resonance parameters
+    resonance_covariance : openmc.data.ResonanceCovariance or None
+        Covariance for resonance parameters
+    temperatures : list of str
+        List of string representations the temperatures of the target nuclide
```

Comment:
Nit: Did you mean to say "List of string representations **of** the temperatures" rather than "List of string representations the temperatures"?